worker/request_handlers.go

```diff
@@ -2,14 +2,21 @@ package worker
 
 import (
 	"fmt"
-	"github.com/joschahenningsen/TUM-Live-Worker-v2/cfg"
-	"github.com/joschahenningsen/TUM-Live-Worker-v2/pb"
-	log "github.com/sirupsen/logrus"
 	"os/exec"
 	"strings"
 	"time"
+
+	"github.com/joschahenningsen/TUM-Live-Worker-v2/cfg"
+	"github.com/joschahenningsen/TUM-Live-Worker-v2/pb"
+	log "github.com/sirupsen/logrus"
 )
 
+// lectureHallStreams contains a map from streaming keys to their contexts.
+// Note that we can have multiple contexts for different sources.
+var lectureHallStreams = make(map[uint32][]*StreamContext)
+
+var shouldDiscardVoD = make(map[uint32]bool)
```

Comment:
Instead of a new map, you could put this into the `StreamContext`

Reply:
Using `StreamContext` also seems like a good idea! I will add this.